src/NUnitFramework/framework/Internal/Randomizer.cs

```diff
@@ -22,47 +22,72 @@
 // ***********************************************************************
 
 using System;
+using System.Collections;
 using System.Collections.Generic;
 using System.Reflection;
+using System.Text;
 
 namespace NUnit.Framework.Internal
 {
     /// <summary>
     /// Randomizer returns a set of random _values in a repeatable
-    /// way, to allow re-running of tests if necessary. 
+    /// way, to allow re-running of tests if necessary. It extends
+    /// the .NET Random class, providing random values for a much
+    /// wider range of types.
     /// 
-    /// This class is an internal framework class used for setting up tests. 
-    /// It is used to generate random test parameters, at the time of loading
-    /// the tests. It also generates seeds for use at execution time, when
-    /// creating a RandomGenerator for use by the test.
+    /// The class is used internally by the framework to generate 
+    /// test case data and is also exposed for use by users through 
+    /// the TestContext.Random property.
     /// </summary>
+    /// <remarks>
+    /// For consistency with the underlying Random Type, methods 
+    /// returning a single value use the prefix "Next..." Those
+    /// without an argument return a non-negative value up to
+    /// the full positive range of the Type. Overloads are provided
+    /// for specifying a maximum or a range. Methods that return
+    /// arrays or strings use the prefix "Get..." to avoid 
+    /// confusion with the single-value methods.
+    /// </remarks>
     public class Randomizer : Random
     {
         #region Static Members
 
-        private static Random seedGenerator;
-
-        private static Dictionary<MemberInfo, Randomizer> randomizers = new Dictionary<MemberInfo, Randomizer>();
-
         /// <summary>
         /// Initial seed used to create randomizers for this run
         /// </summary>
         public static int InitialSeed { get; set; }
 
+        // Static Random instance used exclusively for the generation
+        // of seed values for new Randomizers.
+        private static Random _seedGenerator;
+        private static Random SeedGenerator
+        {
+            get
+            {
+                if (_seedGenerator == null)
+                    _seedGenerator = new Random(InitialSeed);
```

Comment:
Since this has to be set in order to load the tests, which we currently do on a single thread, I don't think it's a problem. But I can take a look and see if I can find a place to initialize it explicity - maybe when we set InitialSeed.